test/demangler-tests.ts

```diff
@@ -42,7 +42,9 @@ class DummyCompiler extends BaseCompiler {
     constructor() {
         const env = {
             ceProps: properties.fakeProps({}),
-            compilerProps: () => {},
+            getCompilerPropsForLanguage: () => {
```

Comment:
Did `fakeProps` not work out here? and ditto other places?

Reply:
It's a little complicated

So at some point in the unittests `BaseCompiler` is instantiated, and because of me adding a `compilerProps` that is needed, I encountered that the `compilerProps` calls returned `undefined` instead of the default, this is because a `_.partial` is used to resolve `this.compilerProps` in `BaseCompiler`. It usually didn't matter because they weren't used in the happy paths of the unittests, but the newly added one was used for getting an Execution Environment Class and it was like "No such thing as 'undefined', here's an exception".

https://github.com/compiler-explorer/compiler-explorer/pull/6413/files#diff-23200c2e7936d6738c5056e91063698d8d9db45b30fa99be68e22eed138ee8eaR212

The partial in turn returned a disfunctional function instead of something useful. I added an extra indirection with `getCompilerPropsForLanguage` so we can mock it and directly return a function `(key, def) => def` that does the right thing and return the default that we want to not throw errors.

Reply:
Thanks for the explanation. It's sad we have to do this, I'd love to think of a better way. but I see why